src/Initializer/Parameters/OutputParameters.cpp

```diff
@@ -23,41 +23,28 @@ CheckpointParameters readCheckpointParameters(ParameterReader* baseReader) {
   auto* reader = baseReader->readSubNode("output");
 
   auto enabled = reader->readWithDefault("checkpoint", true);
-  auto readBackend = [&reader](bool enabled) {
-    CheckpointingBackend backend = CheckpointingBackend::DISABLED;
-    if (enabled) {
-      backend = reader->readWithDefaultStringEnum<CheckpointingBackend>(
-          "checkpointbackend",
-          "none",
-          {{"none", CheckpointingBackend::DISABLED},
-           {"posix", CheckpointingBackend::POSIX},
-           {"hdf5", CheckpointingBackend::HDF5},
-           {"mpio", CheckpointingBackend::MPIO},
-           {"mpio_async", CheckpointingBackend::MPIO_ASYNC},
-           {"sionlib", CheckpointingBackend::SIONLIB}});
-    } else {
-      reader->markUnused({"checkpointbackend", "checkpointinterval"});
-    }
-    return backend;
-  };
-  const auto backend = readBackend(enabled);
-  const auto interval = reader->readWithDefault("checkpointinterval", 0.0);
-
+  CheckpointingBackend backend = CheckpointingBackend::DISABLED;
+  double interval = 0.0;
+  std::string fileName = "";
   if (enabled) {
+    interval = reader->readWithDefault("checkpointinterval", 0.0);
     warnIntervalAndDisable(enabled, interval, "checkpoint", "checkpointinterval");
   }
```

Comment:
I'd check for `markUnused` for `checkpointinterval` already here—if `enabled==false`

Reply:
I wrote it this way so that all the unused variables are in one place. Would you rather have it in two different if, else statements? I would change that if it is cleaner.

Reply:
Would be better: what could happen is

* `checkpoint=1`
* `checkpointinterval=-1`

Then we mark only two fields with unused. Compare that with `checkpoint=0`. Then, also `checkpointinterval` is unused.

Reply:
> Would be better: what could happen is
> 
> * `checkpoint=1`
> * `checkpointinterval=-1`
> 
> Then we mark only two fields with unused. Compare that with `checkpoint=0`. Then, also `checkpointinterval` is unused—in addition to the two other fields.

I am sorry, I think I am confused:

lets say checkpoint=1, checkpointinterval=-1, here `warnIntervalAndDisable`  turns `enabled=false` and then all the three fields are marked unused in the next if, else statement. Or did you want to not mark the `checkpointinterval` in this case?

If `checkpoint=0`, both the ifs and ignored anyway and the else statement gets activated.

Or did you mean something else entirely?

Reply:
Yes; you're right about that—it won't really matter here, as the reader right now allows a variable to be both read (i.e. used) and begin marked as unused.

To me, it would come over cleaner if the `checkpointinterval` would be marked as unused in the first `if`, since it's being queried in the other part of the `if` there. Just in case the logic changes at some point in the future.

Otherwise, it would be best to add a comment why `checkpointinterval` is marked as unused only there.